.teamcity/src/main/kotlin/configurations/CompileAll.kt

```diff
@@ -1,30 +1,59 @@
 package configurations
 
+import configurations.CompileAll.BuildCacheType.PRODUCTION_BUILD_CACHE
 import model.CIBuildModel
 import model.Stage
 
-class CompileAll(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(stage = stage, init = {
-    id(buildTypeId(model))
-    name = "Compile All"
-    description = "Compiles all production/test source code and warms up the build cache"
+class CompileAll(model: CIBuildModel, stage: Stage, buildCacheType: BuildCacheType = PRODUCTION_BUILD_CACHE) :
+    BaseGradleBuildType(stage = stage, failStage = buildCacheType.failStage, init = {
+        id(buildTypeId(model, buildCacheType))
+        name = "Compile All${buildCacheType.nameSuffix}"
+        description = "Compiles all production/test source code and warms up the build cache"
 
-    features {
-        publishBuildStatusToGithub(model)
-    }
+        features {
+            publishBuildStatusToGithub(model)
+        }
+
+        applyDefaults(
+            model,
+            this,
+            "compileAllBuild -PignoreIncomingBuildReceipt=true -DdisableLocalCache=true ${buildCacheType.additionalParameters}",
+            extraParameters = buildScanTag("CompileAll") + " " + "-Porg.gradle.java.installations.auto-download=false"
```

Comment:
The build cache ng build shouldn't have the compile all tag. We use that tag in a chain to figure out which build produced the artifacts.